src/topology/algebra/order/lower_topology.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import order.upper_lower
+import topology.separation
+
+/-!
+# Lower topology
+
+This file introduces the lower topology on a preorder. It is shown that the lower topology on a
+partial order is T₀ and the complements of the upper closures of finite subsets form a basis.
+
+## Implementation notes
+
+Approach inspired by `order_topology`.
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+
+variable (α : Type u)
+
+open set topological_space
+
+section preorder
+
+variable [preorder α]
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[derive preorder, nolint unused_arguments]
+def with_lower_topology := α
```

Comment:
Can you copy and adapt the corresponding (light) `lex` machinery?
https://github.com/leanprover-community/mathlib/blob/7a7d4dd424b60b6d18c58afed3aea2dfac64fa16/src/order/synonym.lean#L89-L103

Reply:
@YaelDillies Should it now look something like this?
```
def with_lower_topology := α

instance [p : preorder α] : preorder (with_lower_topology) := p
instance [p : nonempty α] : nonempty (with_lower_topology) := p
instance [p : inhabited α] : inhabited (with_lower_topology) := p

/-- `to_with_lower_topology` is the identity function to the `with_lower_topology` of a type.  -/
@[pattern] def to_with_lower_topology : α ≃ with_lower_topology := equiv.refl _

/-- `of_with_lower_topology` is the identity function from the `with_lower_topology` of a type.  -/
@[pattern] def of_with_lower_topology : with_lower_topology ≃ α := equiv.refl _

@[simp] lemma to_with_lower_topology_symm_eq : (@to_with_lower_topology α).symm = 
  of_with_lower_topology := rfl
@[simp] lemma of_with_lower_topology_symm_eq : (@of_with_lower_topology α).symm = 
  to_with_lower_topology := rfl
@[simp] lemma to_with_lower_topology_of_with_lower_topology (a : @with_lower_topology α) : 
  to_with_lower_topology (of_with_lower_topology a) = a := rfl
@[simp] lemma of_with_lower_topology_to_with_lower_topology (a : α) : of_with_lower_topology
  (to_with_lower_topology a) = a := rfl
@[simp] lemma to_with_lower_topology_inj {a b : α} : to_with_lower_topology a = 
  to_with_lower_topology b ↔ a = b := iff.rfl
@[simp] lemma of_with_lower_topology_inj {a b : @with_lower_topology α} : of_with_lower_topology a 
  = of_with_lower_topology b ↔ a = b := iff.rfl

/-- A recursor for `lex`. Use as `induction x using lex.rec`. -/
protected def with_lower_topology.rec {β : @with_lower_topology α → Sort*} 
  (h : Π a, β (to_with_lower_topology a)) : Π a, β a := λ a, h (of_with_lower_topology a)

variable [preorder α]

/--
The lower topology is the topology generated by the complements of the closed intervals to infinity.
-/
def lower_topology : topological_space α := generate_from {s | ∃ a, (Ici a)ᶜ = s}
```

Reply:
`α` should still be explicit in `with_lower_topology`. `to/of_with_lower_topology` seems too long; I'd go for `to_of/lower`, maybe in the `with_lower_topology` namespace.

The formatting
```
@[simp] lemma of_with_lower_topology_to_with_lower_topology (a : α) : of_with_lower_topology
  (to_with_lower_topology a) = a := rfl
```
is a bit weird to me; I'd go for
```
@[simp] lemma of_lower_to_lower (a : α) :
  of_lower (to_lower a) = a := rfl

@[simp] lemma of_lower_inj {a b : with_lower_topology α} :
  of_lower a = of_lower b ↔ a = b := iff.rfl
```
etc.

Reply:
With the shorter names it all fits on one line.

Reply:
`to_lower`/`of_lower` are too ambiguous. Either lengthen them or dump them in a namespace.

Reply:
they're now in with_lower_topology namespace.